crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/acir_variable.rs

```diff
@@ -793,6 +795,47 @@ impl AcirContext {
             }
         }
     }
+
+    /// Generate output variables that are constrained to be the sorted inputs
+    /// The outputs are the sorted inputs iff
+    /// outputs are sorted and
+    /// outputs are a permutation of the inputs
+    pub(crate) fn sort(
+        &mut self,
+        inputs: Vec<AcirVar>,
+        bit_size: u32,
+    ) -> Result<Vec<AcirVar>, AcirGenError> {
+        let len = inputs.len();
+        // Convert the inputs into expressions
+        let inputs_expr = vecmap(inputs, |input| self.vars[input].to_expression().into_owned());
+        // Generate output witnesses
+        let outputs_witness = vecmap(0..len, |_| self.acir_ir.next_witness_index());
+        let output_expr =
+            vecmap(&outputs_witness, |witness_index| Expression::from(*witness_index));
+        let outputs_var = vecmap(&outputs_witness, |witness_index| {
+            self.add_data(AcirVarData::Witness(*witness_index))
+        });
+        // Enforce the outputs to be sorted
+        for i in 0..(outputs_var.len() - 1) {
+            self.less_than_constrain(outputs_var[i], outputs_var[i + 1], bit_size)?;
+        }
+        // Enforce the outputs to be a permutation of the inputs
+        self.acir_ir.permutation(&inputs_expr, &output_expr);
+
+        Ok(outputs_var)
+    }
+
+    /// Constrain lhs to be less than rhs
+    fn less_than_constrain(
```

Comment:
This is pretty confusing, given that we have `less_than_var` -- why can't you use that method?

Reply:
Done